resolve issue of operations drowpdown assigned to all after reloading SearchTracePage SearchForm page.

het4399 opened the pull request:
## Which problem is this PR solving?
Resolves #2599 
## Description of the changes
- Ensure SearchForm only renders when service operations are loaded and always include operationsForService in props.

## How was this change tested?
- 2 tests are failing right now, but wanted feedback on the approach before spending time on fixing tests.
   - <SearchTracePage shows a search form when services are loaded
   -  mapStateToProps() › converts state to the necessary props

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `npm run lint` and `npm run test`

yurishkuro commented on packages/jaeger-ui/src/components/SearchTracePage/index.jsx:
```diff
@@ -106,15 +106,27 @@ export class SearchTracePageImpl extends Component {
+        tabItems.push({ label: 'Search', key: 'searchForm', children: <SearchForm services={services} /> });
+      }
     } else {
       tabItems.push({ label: 'Search', key: 'searchForm', children: <LoadingIndicator /> });
```
can we reduce repetition?

het4399 commented:
@yurishkuro I've removed the previous logic and refactored the stateServicesXformer function to the handle  loading state for operations and all tests are now passing.

yurishkuro commented:
solved in #2995
